log4j-core/src/main/java/org/apache/logging/log4j/core/util/internal/InternalLoggerRegistry.java

```diff
@@ -147,44 +147,43 @@ public Logger computeIfAbsent(
             return logger;
         }
 
+        // Creating a logger is expensive and might cause lookups and locks, possibly deadlocks:
+        // https://github.com/apache/logging-log4j2/issues/3252
+        // https://github.com/apache/logging-log4j2/issues/3399
+        //
+        // Creating loggers without a lock, allows multiple threads to create loggers in parallel, which also
+        // improves performance.
+        // Since all loggers with the same parameters are equivalent, we can safely return the logger from the
+        // thread that finishes first.
+        Logger newLogger = loggerSupplier.apply(name, messageFactory);
```

Comment:
_Nit:_ I prefer the old `logger = loggerSupplier.apply()`, this makes it clear that the variable names sharing the same prefix and introduced further down (`loggerName`, `loggerMessageFactory`, etc.) are of same origin.